Applications/SimpleShot/layers/centorid_nearest_neighbors.cpp

```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2020 Jihoon Lee <jhoon.it.lee@samsung.com>
+ *
+ * @file   nearest_neighbors.cpp
+ * @date   09 Jan 2021
+ * @brief  This file contains the simple nearest neighbors layer, this layer
+ * takes centroid and calculate l2norm
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Jihoon Lee <jhoon.it.lee@samsung.com>
+ * @bug    No known bugs except for NYI items
+ *
+ */
+
+#include <iostream>
+#include <limits>
+#include <regex>
+#include <sstream>
+
+#include <nntrainer_error.h>
+#include <nntrainer_log.h>
+#include <tensor.h>
+#include <weight.h>
+
+#include <centroid_nearest_neighbors.h>
+#include <simpleshot_utils.h>
+
+namespace simpleshot {
+namespace layers {
+
+const std::string CentroidNearestNeighborsLayer::type = "centroid_nn";
+
+int CentroidNearestNeighborsLayer::setProperty(
+  std::vector<std::string> values) {
+  util::Entry e;
+
+  std::vector<std::string> unhandled_values;
+
+  for (auto &val : values) {
+    try {
+      e = util::getKeyValue(val);
+    } catch (std::invalid_argument &e) {
+      std::cerr << e.what() << std::endl;
+      return -1;
+    }
+
+    if (e.key == "num_class") {
+      try {
+        num_class = std::stoi(e.value);
+      } catch (std::invalid_argument &e) {
+        std::cerr << e.what() << std::endl;
+        return -1;
+      } catch (std::out_of_range &e) {
+        std::cerr << e.what() << std::endl;
+        return -1;
+      }
+    } else {
+      unhandled_values.push_back(val);
+    }
+  }
+
+  return nntrainer::Layer::setProperty(unhandled_values);
+}
+
+int CentroidNearestNeighborsLayer::initialize(nntrainer::Manager &manager) {
+  if (input_dim[0].channel() || input_dim[0].height()) {
+    ml_logw("centroid nearest layer is designed for flattend feature for now, "
+            "please check");
+  }
+
+  if (num_class == 0) {
+    ml_loge("centroid nearest layer needs to define num_class for now, please "
```

Comment:
"centroid nearest layer needs to define num_class for now, please check" -> "Error: num_class must be a positive non-zero integer."

Reply:
Done